### DISCUSSION TITLE
@vercel/postgres SELECT queries WHERE clause not handling values with symbols

### DISCUSSION THREAD

**Discussion by amroczeK** [vercel/community#2938] [Help] [answered]
### Summary

I am having trouble doing the following select statement using `import { sql } from "@vercel/postgres";` in my NextJs project.

Via CLI (psql) with the following command: 
`SELECT * FROM Guests WHERE address = '2/105 Seventh Ave, Maylands WA, Australia';` I get this result:
![image](https://github.com/vercel/community/assets/18630966/6f8b8f59-981d-4491-9ae7-a656ee11cf69)

I am unable to use the vercel postgres npm package to retrieve this data programattically in nextjs. I have tried the following examples in NextJs API route with no luck:

1. Following documented example https://vercel.com/docs/storage/vercel-postgres/sdk#sql
```
const { rows } =
      await sql`SELECT * FROM Guests WHERE address = ${address};`;
```

2. Adding apostophes around ${address} line in psql returns the following error:
```
const { rows } = await sql`SELECT * FROM Guests WHERE address = '${address}';`;
```
> error: bind message supplies 1 parameters, but prepared statement "" requires 0

3. Wrapping ${address} in double quotations returns this error:
```
const { rows } = await sql`SELECT * FROM Guests WHERE address = "${address}";`;
```
> error: column "$1" does not exist

4. Adding the address directly with apostohpes works and returns results
```
const { rows } = await sql`SELECT * FROM Guests WHERE address = '2/105 Seventh Ave, Maylands WA, Australia';`;
```

5. I have tried:
```
const query = "SELECT * FROM Guests WHERE address = '" + address + "';";
console.log(query); // SELECT * FROM Guests WHERE address = '2/105 Seventh Ave, Maylands WA, Australia';
const { rows } = await sql`${query}`;
```

What is the proper way to use `sql` from the SDK to query a table for a row that has a value which has a symbol e.g. addresses with a unit number?

I have also tried:
- Encoding the address
- Using escape squences
- Tried PSQL string functions

### Steps to Reproduce

1. Go to Vercel -> Storage and create a postgresql database
2. Create a table with a column called 'address'
3. Insert one row and populate the address column with a value of an address which is a unit/apartment e.g. 12/110 Mounts Bay Road,Perth, WA 6000.
4. Try and retrieve that data using sql exposed by `import { sql } from "@vercel/postgres";` example docs: https://vercel.com/docs/storage/vercel-postgres/sdk

**Comment by amyegan** [comment]
I wasn't able to reproduce the issue in my test app. How are is `address` being assigned before it is used in the sql query?

**Reply by amroczeK** [reply, marked as answer]
Hey @amyegan I am closing this discussion because I created a duplicate issue here: https://github.com/vercel/storage/issues/229

I have detailed what the potential cause was there.
